src/einsteinpy/coordinates/core.py

```diff
@@ -217,6 +263,14 @@ def __init__(self, r, theta, phi, a):
         self.a = a
         super().__init__(r.si.value, theta.si.value, phi.si.value, a=a.si.value)
         self.system = "BoyerLindquist"
+        self._dimension = {
+            "r": self.r,
+            "theta": self.theta,
+            "phi": self.phi,
+            "a": self.a,
+            "system": "BoyerLindquist",
+        }
+        self._dimension_order = ("r", "theta", "phi", "a")
```

Comment:
"a" is not a dimension here. It is an attribute of space-time. Can you remove it from `_dimension_order`?

Reply:
P.S. : feature of the space spanned by BL coordinates

Reply:
This one thing, and then we can rebase!